### ISSUE TITLE
Non null-safe assignment when dealing with wildcards

### ISSUE THREAD

**Issue by theosotr** [typetools/checker-framework#6881]
### Command

```
javac -J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
-processor org.checkerframework.checker.nullness.NullnessChecker \
-cp checker-framework/checker/dist/checker.jar -Aignorejdkastub Test.java
```

### File

```java
import java.util.*;
import org.checkerframework.checker.nullness.qual.*;

public class Test {

  static public void main(String[] args) {
      List<List<? extends Number>> lists = new ArrayList<>();
      List<? super List<? extends Number>> y = lists;
      // This assignment is not null-safe
      List<? super @Nullable List<? extends Number>> x = y;
      x.add(null);
      lists.get(0).toString();
  }
}
```

### Actual behavior

The code passes the checks of the checker, but there's NPE

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "java.util.List.get(int)" is null
        at Test.main(Test.java:12)
```

### Expected behavior

The assignment `List<? super @Nullable List<? extends Number>> x = y;` should have been rejected.
